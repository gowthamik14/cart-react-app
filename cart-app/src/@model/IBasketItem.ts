import IProduct from './IProduct';

export default interface IBasketItem extends IProduct {
  quantity: number;
}
