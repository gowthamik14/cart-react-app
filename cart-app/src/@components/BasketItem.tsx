import React from 'react';
import IBasketItem from '@model/IBasketItem';
import Quantity from './Quantity';
import BasketItemCost from './BasketItemCost';

interface BasketItemProp {
  basketItem: IBasketItem;
  savingAmount: number;
  increaseQuantity: (basketItem: IBasketItem) => void;
  decreaseQuantity: (basketItem: IBasketItem) => void;
}

const BasketItem: React.FC<BasketItemProp> = ({
  basketItem,
  savingAmount,
  increaseQuantity,
  decreaseQuantity,
}) => {
  return (
    <li
      className="flex flex-col list-decimal list-inside border-b-2 border-gray-100	p-2"
      key={basketItem.id}
    >
      <div className="flex justify-between items-center">
        <span className="text-lg">{basketItem.name}</span>
        <div>
          <span className="text-gray-400 pr-1">Â£</span>
          <span className="text-lg">{(basketItem.price / 100).toFixed(2)}</span>
        </div>
        <Quantity
          item={basketItem}
          increaseQuantity={increaseQuantity}
          decreaseQuantity={decreaseQuantity}
        />
      </div>
      <div>
        <BasketItemCost
          price={basketItem.price}
          quantity={basketItem.quantity}
          savingAmount={savingAmount}
        />
      </div>
    </li>
  );
};
export default BasketItem;
