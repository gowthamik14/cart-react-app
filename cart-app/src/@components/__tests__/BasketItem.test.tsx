import React from 'react';
import { render } from '@testing-library/react';
import BasketItem from '@components/BasketItem';
import IBasketItem from '@model/IBasketItem';

test('renders learn react link', () => {
  const basketItem: IBasketItem = {
    name: 'cheese',
    price: 20,
    quantity: 3,
    id: 1,
  };
  const savingAmount: number = 20;
  const increaseQuantity = jest.fn();
  const decreaseQuantity = jest.fn();
  const { container } = render(
    <BasketItem
      basketItem={basketItem}
      savingAmount={savingAmount}
      increaseQuantity={increaseQuantity}
      decreaseQuantity={decreaseQuantity}
    />
  );


  expect(container).toMatchSnapshot();
});
