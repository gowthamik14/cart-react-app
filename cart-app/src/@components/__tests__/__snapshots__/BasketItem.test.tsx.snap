// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div>
  <li
    class="flex flex-col list-decimal list-inside border-b-2 border-gray-100	p-2"
  >
    <div
      class="flex justify-between items-center"
    >
      <span
        class="text-lg"
      >
        cheese
      </span>
      <div>
        <span
          class="text-gray-400 pr-1"
        >
          £
        </span>
        <span
          class="text-lg"
        >
          0.20
        </span>
      </div>
      <span>
        <button
          class="ml-4 p-2 bg-blue-400 text-white rounded-md focus:outline-none focus:ring-2 ring-blue-200 ring-offset-2"
        >
          +
        </button>
        <span
          class="ml-4 p-2 w-10 text-center text-blue rounded-md"
        >
          3
        </span>
        <button
          class="ml-4 p-2  bg-white-400 text-blue rounded-md border-2 border-blue-400 focus:outline-blue focus:ring-2 ring-blue-200 ring-offset-2"
        >
          -
        </button>
      </span>
    </div>
    <div>
      <div
        class="flex flex-col justify-center"
      >
        <div
          class="flex justify-end my-2 border-b-2 border-gray-100 text-gray-500"
        >
          <div
            class="mx-2"
          >
            Item price 
          </div>
          <div>
            £0.20 * 3 = £0.60
          </div>
        </div>
        <div
          class="flex justify-end items-center my-2 border-b-2 border-gray-100 text-red-500"
        >
          <div
            class="mx-2"
          >
            Savings
          </div>
          <div>
            £0.20
          </div>
        </div>
        <div
          class="flex justify-end my-2 items-center"
        >
          <div
            class="mx-2"
          >
            Item cost 
          </div>
          <div>
            £0.40
          </div>
        </div>
      </div>
    </div>
  </li>
</div>
`;
